### PR TITLE
Add follow-up date as a column to all cases view

### PR DESCRIPTION
**Pull request by GPaoloni**
Jira: https://bugs.benetech.org/browse/CHI-388

Primary reviewer: @andvirga 

This PR:
- Adds "follow up date" as a new column in All Cases list.
- "Updated" field to match the same behavior as the "Last Updated/Closed" field in Case View.
- Moves connection to Redux store from every child case component to the parent one.

### CODE REVIEW

File: plugin-hrm-form/src/components/caseList/CaseListTableRow.jsx [hunk 1]
```diff
@@ -46,7 +44,12 @@ const CaseListTableRow = ({ caseItem, counselorsHash, openMockedMessage }) => {
   const shortSummary = getShortSummary(summary, CHAR_LIMIT, 'case');
   const counselor = formatName(counselorsHash[caseItem.twilioWorkerId]);
   const opened = `${format(new Date(caseItem.createdAt), 'MMM d, yyyy')}`;
-  const updated = `${format(new Date(caseItem.updatedAt), 'MMM d, yyyy')}`;
+  const beenUpdated = caseItem.createdAt !== caseItem.updatedAt;
```
Comment (nickhurlburt) [inline plugin-hrm-form/src/components/caseList/CaseListTableRow.jsx:47]: `caseItem.createdAt` and `caseItem.updatedAt` are dates with hours/minutes/seconds, yes?  I think if the case was updated later in the day, we want to display that day, even if it is the same.

**Review by nickhurlburt — APPROVED** [review]
Looks great!

Note there are two console warnings that come up when loading All Cases View:
```
react_devtools_backend.js:2430 Warning: Failed prop type: Invalid prop `caseItem.info` of type `object` supplied to `CaseListTableRow`, expected `string`.
    in CaseListTableRow (at CaseListTable.jsx:34)
    in CaseListTable (created by Connect(CaseListTable))
    in Connect(CaseListTable) (at CaseList.jsx:94)
    in div (created by CaseListContainer)
    in CaseListContainer (at CaseList.jsx:93)
    in CaseList (at setUpComponents.js:363)
```

and

```
Warning: Received `false` for a non-boolean attribute `transparent`.

If you want to write it to the DOM, pass a string instead: transparent="false" or transparent={value.toString()}.

If you used to conditionally omit it with transparent={condition && value}, pass transparent={condition ? value : undefined} instead.
    in tr (created by Context.Consumer)
    in TableRow (created by t)
    in t (created by PaginationRow)
    in PaginationRow (at Pagination.tsx:79)
    in tfoot (created by TableFooter)
    in TableFooter (created by t)
    in t (at Pagination.tsx:78)
    in Pagination (at CaseListTable.jsx:42)
    in table (created by Table)
    in Table (created by t)
    in t (created by CLTable)
    in CLTable (at CaseListTable.jsx:30)
    in div (created by TableContainer)
    in TableContainer (at CaseListTable.jsx:29)
    in CaseListTable (created by Connect(CaseListTable))
    in Connect(CaseListTable) (at CaseList.jsx:94)
```

You probably didn't cause these, but if they are easy fixes, please do so.  Our console logs are getting more polluted and it's worth a few minutes to clean that up.

File: plugin-hrm-form/src/components/caseList/CaseListTableRow.jsx (hunk 1 above)
Comment (GPaoloni) [reply plugin-hrm-form/src/components/caseList/CaseListTableRow.jsx:47]: So, if it's updated a few minutes later, we want to show the updated date anyway? Or in that case we show "-"?

Comment (nickhurlburt) [reply plugin-hrm-form/src/components/caseList/CaseListTableRow.jsx:47]: If it's been updated at all, yes we want to show the updated date.

File: plugin-hrm-form/src/components/caseList/CaseListTableRow.jsx [hunk 2]
```diff
@@ -110,8 +119,4 @@ CaseListTableRow.propTypes = {
   openMockedMessage: PropTypes.func.isRequired,
 };
 
-const mapStateToProps = state => ({
-  counselorsHash: state[namespace][configurationBase].counselors.hash,
-});
-
-export default connect(mapStateToProps)(CaseListTableRow);
```
Comment (andvirga) [inline plugin-hrm-form/src/components/caseList/CaseListTableRow.jsx:117]: Nicely done!

**Review by andvirga — APPROVED** [review]
Great job!

**Pull request merged**